### PR TITLE
[Impeller] libImpeller: Document the entire API.

### PR DESCRIPTION
**Pull request by chinmaygarde**
I believe I got the entire API other than the enum values. Though most a straightforward, some could use some clarification.

The typography stuff should now be easier to use. The README describes the object model and reference counting conventions.

### CODE REVIEW

**Comment by chinmaygarde** [comment]
cc @lyceel. I believe I have documented all the issues that came up. Happy to elaborate where necessary.

File: impeller/toolkit/interop/impeller.h [hunk 1]
```diff
@@ -584,44 +1354,130 @@ ImpellerColorFilterCreateColorMatrixNew(
...
+///
+/// @return     The image filter.
+///
 IMPELLER_EXPORT IMPELLER_NODISCARD ImpellerImageFilter IMPELLER_NULLABLE
 ImpellerImageFilterCreateMatrixNew(
     const ImpellerMatrix* IMPELLER_NONNULL matrix,
     ImpellerTextureSampling sampling);
 
+//------------------------------------------------------------------------------
+/// @brief      Creates a combo filter that when applied is identical to
```
Comment (jonahwilliams) [inline impeller/toolkit/interop/impeller.h:1469]: nit: combo -> composed?

File: impeller/toolkit/interop/impeller.h [hunk 2]
```diff
@@ -719,25 +1704,54 @@ void ImpellerDisplayListBuilderRestoreToCount(
...
+/// @param[in]  op           The operation.
+///
 IMPELLER_EXPORT
 void ImpellerDisplayListBuilderClipOval(
     ImpellerDisplayListBuilder IMPELLER_NONNULL builder,
     const ImpellerRect* IMPELLER_NONNULL oval_bounds,
     ImpellerClipOperation op);
 
+//------------------------------------------------------------------------------
+/// @brief      Clip away the contents outside the specified shape.
```
Comment (jonahwilliams) [inline impeller/toolkit/interop/impeller.h:1734]: This depends on the clip op though.

**Review by jonahwilliams — APPROVED** [review]
LGTM with nits

File: impeller/toolkit/interop/impeller.h (hunk 2 above)
Comment (chinmaygarde) [reply impeller/toolkit/interop/impeller.h:1734]: Done.

File: impeller/toolkit/interop/impeller.h (hunk 1 above)
Comment (chinmaygarde) [reply impeller/toolkit/interop/impeller.h:1469]: Done.

**Pull request merged**